unitTests/main.cpp

```diff
@@ -862,27 +742,29 @@ TEST_F(STDTypeTraits, std_common_type)
 
 TEST_F(STDSpecializations, hash)
 {
-	EXPECT_EQ(std::hash<meter_t<double>>()(3.14_m), std::hash<double>()(3.14));
-	EXPECT_EQ(std::hash<millimeter_t<double>>()(3.14_m), std::hash<double>()(3.14e3));
-	EXPECT_EQ(std::hash<millimeter_t<double>>()(3.14_mm), std::hash<double>()(3.14));
-	EXPECT_EQ(std::hash<kilometer_t<double>>()(3.14_m), std::hash<double>()(3.14e-3));
-	EXPECT_EQ(std::hash<kilometer_t<double>>()(3.14_km), std::hash<double>()(3.14));
+	EXPECT_EQ(std::hash<meter<double>>()(3.14_m), std::hash<double>()(3.14));
+	EXPECT_EQ(std::hash<millimeter<double>>()(3.14_m), std::hash<double>()(3.14e3));
+	EXPECT_EQ(std::hash<millimeter<double>>()(3.14_mm), std::hash<double>()(3.14));
+	EXPECT_EQ(std::hash<kilometer<double>>()(3.14_m), std::hash<double>()(3.14e-3));
+	EXPECT_EQ(std::hash<kilometer<double>>()(3.14_km), std::hash<double>()(3.14));
 
-	EXPECT_EQ((std::hash<unit<meters, int>>()(unit<meters, int>(42))), 42);
-	EXPECT_EQ((std::hash<unit<millimeters, int>>()(unit<meters, int>(42))), 42000);
-	EXPECT_EQ((std::hash<unit<millimeters, int>>()(unit<millimeters, int>(42))), 42);
-	EXPECT_EQ((std::hash<unit<kilometers, int>>()(unit<kilometers, int>(42))), 42);
+	EXPECT_EQ((std::hash<meter<int>>()(meter<int>(42))), 42);
+	EXPECT_EQ((std::hash<millimeter<int>>()(meter<int>(42))), 42000);
+	EXPECT_EQ((std::hash<millimeter<int>>()(millimeter<int>(42))), 42);
+	EXPECT_EQ((std::hash<kilometer<int>>()(kilometer<int>(42))), 42);
 
 	EXPECT_EQ((std::hash<dimensionless<double>>()(3.14)), std::hash<double>()(3.14));
-	EXPECT_EQ((std::hash<unit<dimensionless_unit, double>>()(3.14)), std::hash<double>()(3.14));
-	EXPECT_EQ((std::hash<dimensionless<int>>()(42)), (std::hash<unit<dimensionless_unit, int>>()(42)));
+	EXPECT_EQ((std::hash<dimensionless<double>>()(3.14)), std::hash<double>()(3.14));

```

Comment:
Same here. You'll see that the line above is now equal to this one. And below, too, at the tests for the ordering operators.